src/include/storage/storage_defs.h

```diff
@@ -43,16 +43,15 @@ struct BlockLayout {
    * attr_sizes, as the constructor applies optimizations based on sizes. It is up to the caller to then
    * associate these "column ids" with the right upper level concepts.
    *
-   * @param num_attrs number of attributes.
    * @param attr_sizes vector of attribute sizes.
    */
-  BlockLayout(const uint16_t num_attrs, std::vector<uint8_t> attr_sizes)
-      : num_cols_(num_attrs),
+  explicit BlockLayout(std::vector<uint8_t> attr_sizes)
+      : num_cols_(static_cast<uint16_t>(attr_sizes.size())),
```

Comment:
This field is probably also redundant. I believe I went through the code and removed any direct reference to it already, so it should be easy to remove this as well?